op-proposer/proposer/driver.go

```diff
@@ -68,42 +71,78 @@ type L2OutputSubmitter struct {
 
 	l2ooContract *bindings.L2OutputOracleCaller
 	l2ooABI      *abi.ABI
+
+	dgfContract *bindings.DisputeGameFactoryCaller
+	dgfABI      *abi.ABI
 }
 
 // NewL2OutputSubmitter creates a new L2 Output Submitter
 func NewL2OutputSubmitter(setup DriverSetup) (*L2OutputSubmitter, error) {
 	ctx, cancel := context.WithCancel(context.Background())
 
```

Comment:
Why can't you `defer cancel()` here, and omit all the `cancel()` calls in error paths?

Reply:
I think because the `L2OutputSubmitter` is passed the `cancel` handle when there's no error